wpimath/src/main/java/edu/wpi/first/wpilibj/kinematics/SwerveModuleState.java

```diff
@@ -48,4 +48,23 @@ public String toString() {
     return String.format(
         "SwerveModuleState(Speed: %.2f m/s, Angle: %s)", speedMetersPerSecond, angle);
   }
+
+  /**
+   * Optimize a desired state by deciding weather to reverse the direction the wheel spins in order
+   * to minimize travel time. This means that the furthest a wheel will ever rotate is 90 degrees,
+   * if used in conjunction with the PIDController class' continuous input functionality.
+   *
+   * @param desiredState The desired state.
+   * @param currentAngle The current module angle.
+   */
+  public static SwerveModuleState optimize(
+      SwerveModuleState desiredState, Rotation2d currentAngle) {
+    var delta = desiredState.angle.minus(currentAngle);
+    if (Math.abs(delta.getDegrees()) > 90.0) {
+      return new SwerveModuleState(
+          desiredState.speedMetersPerSecond * -1.0,
+          desiredState.angle.rotateBy(Rotation2d.fromDegrees(180.0)));
+    }
+    return new SwerveModuleState(desiredState.speedMetersPerSecond, desiredState.angle);
```

Comment:
Should this be in the else branch?

Reply:
https://discord.com/channels/176186766946992128/368993897495527424/797283205296160798

Reply:
I bring this up because it's a bit unusual to write the code as if it was an if-else, but then just not write the else.